flytekit/types/structured/basic_dfs.py

```diff
@@ -165,7 +169,9 @@ def decode(
         ctx: FlyteContext,
         flyte_value: literals.StructuredDataset,
         current_task_metadata: StructuredDatasetMetadata,
-    ) -> pa.Table:
+    ) -> "pa.Table":
```

Comment:
Importing modules can be time-consuming, which is a big benefit of the `TYPE_CHECKING` guard.

I'm not sure if the forward reference here was necessary to appease the type checker, but some older versions of Python would require a forward reference where newer ones would not. Another thing you can do is add `from __future__ import annotations`.

Reply:
In the long term, we can not rely on `from __future__ import annotations`. When [PEP 649](https://peps.python.org/pep-0649/) gets introduced in Python 3.13, it will supersede and deprecate [PEP 563](https://peps.python.org/pep-0563/). PEP 563 introduced `from __future__ import annotations`